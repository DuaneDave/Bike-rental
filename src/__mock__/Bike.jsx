import React, { useState } from "react";


const Bike = () => {

  const [color, setColor] = useState("red");

  const changeColor = (color) => {
    setColor(color);
  };



  const bike = {
    id: 1,
    name: "Bike",
    brand: "Brand",
    bike_type: "Type",
    daily_rate: 100,
    description: "Description",
    color: ["red", "blue", "green"],
    images: {
      red: "https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
      blue: "https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
      green: "https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
    },

  };


  const styles = {
    detailSide: "w-50",
  };

  return (
    <section >
      <div className="grid">
        <div >
          <img
            onClick={() => navigate(`/bikes/${bike.id}`)}
            src={bike.images[color] || bike.images[Object.keys(bike.images)[0]]}
          />
          <span>{bike.description}</span>
        </div>
        <div className={styles.detailSide + " flex flex-column"}>
          <h2>
            <strong>
              {bike.name} {bike.brand}
            </strong>
          </h2>
          <div>
            <span className="flex gap">
              <p>Type</p> {bike.bike_type}
            </span>
            <span className="flex gap">
              <p>Model</p> {bike.brand}
            </span>
            <span className="flex gap">
              <p>Price</p> ${bike.daily_rate}
            </span>
            <span className="flex gap">
              <p>Duration</p> 9 Months
            </span>
          </div>
          <h3>
            <strong>6.7% APR</strong> Representative
          </h3>
          <div className="flex gap">
            {bike.color.map((color) => (
              <p
                key={color}
                changeColor={changeColor}
                displayColor={color}
              ></p>
            ))}
          </div>
          <button to="/reserve" state={{ id: bike.id }}>
            Reserve
          </button>
        </div>
      </div>
    </section>
  );
};

export default Bike;
