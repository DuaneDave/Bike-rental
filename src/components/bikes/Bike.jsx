import ColorSwitch from "../colorSwitch/ColorSwitch";
import { useState } from "react";
import Size from "./Size/Size";
import { Link, useNavigate } from "react-router-dom";

import styles from "./Bike.module.css";
import Button from "../../reusables/button/Button";

function Bike({ bike }) {
  const [color, setColor] = useState(null);
  const changeColor = (input) => setColor(input);

  const navigate = useNavigate();

  return (
    <section className={styles.detailsContainer + " flex center"}>
      <div className="grid">
        <div className={styles.detailMain}>
          <img
            onClick={() => navigate(`/bikes/${bike.id}`)}
            src={bike.images[color] || bike.images[Object.keys(bike.images)[0]]}
          />
          <span>{bike.description}</span>
        </div>
        <div className={styles.detailSide + " flex flex-column"}>
          <h2>
            <strong>
              {bike.name} {bike.brand}
            </strong>
          </h2>
          <div>
            <span className="flex gap">
              <p>Type</p> {bike.bike_type}
            </span>
            <span className="flex gap">
              <p>Model</p> {bike.brand}
            </span>
            <span className="flex gap">
              <p>Price</p> ${bike.daily_rate}
            </span>
            <span className="flex gap">
              <p>Duration</p> 9 Months
            </span>
          </div>
          <h3>
            <strong>6.7% APR</strong> Representative
          </h3>
          <Size />
          <div className="flex gap">
            {bike.color.map((color) => (
              <ColorSwitch
                key={color}
                changeColor={changeColor}
                displayColor={color}
              />
            ))}
          </div>
          <Link
            className={styles.res}
            to="/reserve"
            state={{ id: bike.id, name: bike.name, brand: bike.brand }}
          >
            Reserve
          </Link>
        </div>
      </div>
    </section>
  );
}

export default Bike;
