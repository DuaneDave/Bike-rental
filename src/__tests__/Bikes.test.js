import React from 'react';
import renderer from 'react-test-renderer';
import Bike from '../__mock__/Bike';

describe('Bikes', () => {
  test('if bikes mocked automatically', () => {
    const component = renderer.create(
      <Bike />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
