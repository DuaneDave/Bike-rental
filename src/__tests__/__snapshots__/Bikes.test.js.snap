// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bikes if bikes mocked automatically 1`] = `
<section>
  <div
    className="grid"
  >
    <div>
      <img
        onClick={[Function]}
        src="https://images.unsplash.com/photo-1610000000000-000000000000?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"
      />
      <span>
        Description
      </span>
    </div>
    <div
      className="w-50 flex flex-column"
    >
      <h2>
        <strong>
          Bike
           
          Brand
        </strong>
      </h2>
      <div>
        <span
          className="flex gap"
        >
          <p>
            Type
          </p>
           
          Type
        </span>
        <span
          className="flex gap"
        >
          <p>
            Model
          </p>
           
          Brand
        </span>
        <span
          className="flex gap"
        >
          <p>
            Price
          </p>
           $
          100
        </span>
        <span
          className="flex gap"
        >
          <p>
            Duration
          </p>
           9 Months
        </span>
      </div>
      <h3>
        <strong>
          6.7% APR
        </strong>
         Representative
      </h3>
      <div
        className="flex gap"
      >
        <p
          changeColor={[Function]}
          displayColor="red"
        />
        <p
          changeColor={[Function]}
          displayColor="blue"
        />
        <p
          changeColor={[Function]}
          displayColor="green"
        />
      </div>
      <button
        state={
          {
            "id": 1,
          }
        }
        to="/reserve"
      >
        Reserve
      </button>
    </div>
  </div>
</section>
`;
